data_model/src/trigger.rs

```diff
@@ -0,0 +1,348 @@
+/// it's the end-user's responsibility to either unregister the
+/// `Unlimited` variant.
+///
+/// # Considerations
+///
+/// The granularity might not be sufficient to run an action exactly
+/// `n` times. In order to ensure that it is even possible to run the
+/// triggers without gaps, the `Executable` wrapped in the action must
+/// be run before any of the ISIs are pushed into the queue of the
+/// next block.
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, Serialize, Deserialize, IntoSchema)]
+pub struct EventAction {
+    /// The executable linked to this action
+    pub executable: Executable,
+    /// The repeating scheme of the action. It's kept as part of the
+    /// action and not inside the [`Trigger`] type, so that further
+    /// sanity checking can be done.
+    pub repeats: Repeats,
+    /// Technical account linked to this trigger. The technical
+    /// account must already exist in order for `Register<Trigger>` to
+    /// work.
+    pub technical_account: super::account::Id,
+    /// Event filter to identify events on which this action should be performed.
+    pub filter: EventFilter,
+}
+
+impl EventAction {
+    /// Construct an action given `executable`, `repeats`, `technical_account` and `filter`.
+    pub fn new(
+        executable: impl Into<Executable>,
+        repeats: impl Into<Repeats>,
+        technical_account: super::account::Id,
+        filter: EventFilter,
+    ) -> EventAction {
+        EventAction {
+            executable: executable.into(),
+            repeats: repeats.into(),
+            // TODO: At this point the technical account is meaningless.
+            technical_account,
+            filter,
+        }
+    }
+}
+
+impl PartialOrd for EventAction {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for EventAction {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // Exclude the executable. When debugging and replacing
+        // the trigger, its position in Hash and Tree maps should
+        // not change depending on the content.
+        match self.repeats.cmp(&other.repeats) {
+            Ordering::Equal => self.technical_account.cmp(&other.technical_account),
+            ord => ord,
+        }
+    }
+}
+
+impl ExecutionInfo for EventAction {
+    fn executable(&self) -> &Executable {
+        &self.executable
+    }
+
+    fn technical_account(&self) -> &super::account::Id {
+        &self.technical_account
+    }
+}
+
+/// Action that happens every set time or exactly at set time.
+/// If the trigger must be run every time interval, that's the end-user
+/// responsibility to unregister it
+///
+/// # Considerations
+///
+/// It's not guaranteed that this action execution will be done in exactly the same time,
+/// which is specified, but it's gonna be as close as possible.
+/// Is'
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, Serialize, Deserialize, IntoSchema)]
+pub struct TimeAction {
```

Comment:
Another option was to make an `EventFilter` that matches on a specific time stamp. You would not need two separate types then. 

This works just as well, since scoping doesn't exactly apply to timed triggers.